src/ros_vision_interaction/scripts/ros_reading_evaluator.py

```diff
@@ -0,0 +1,121 @@
+        self._is_record_subscriber = rospy.Subscriber(
+            is_record_evaluation_topic,
+            Bool,
+            callback=self.reading_analyzer_callback,
+            queue_size=1
+        )
+
+        self._state_database = statedb
+        self._reading_evaluator = reading_evaluator
+
+    def reading_analyzer_callback(self, is_record):
+        if not is_record.data:
+            rospy.loginfo("Evaluator callback starting")
+            audio_file = self.find_audio_file(
+                self._source_directory,
+                self._extension,
+                self._file_prefix
+            )
+            reading_time = self._reading_evaluator.get_total_speaking_time(audio_file)
+            rospy.loginfo(f"Reading time: {reading_time}")
+            reading_eval_index = self._state_database.get("reading eval index")
+            try:
+                num_of_words = self._state_database.get("reading eval data")[reading_eval_index]["word count"]
+            except IndexError or KeyError:
+                rospy.logerr(f"Reading task data not found for index:{reading_eval_index}")
+                num_of_words = 0
+            try:
+                reading_speed = num_of_words / reading_time
+                rospy.loginfo("Reading speed: {}".format(reading_speed))
+            except ZeroDivisionError:
+                rospy.logerr("Reading time is 0, setting reading speed to 0")
+                reading_speed = 0
+            self._state_database.set("current eval score", reading_speed)
+
+    def find_audio_file(
+            self,
+            directory,
+            extension,
+            file_prefix
+    ):
+        """Finds a file based on the directory, location, and file prefix.
+        Returns the first occurrence of a valid file path."""
+        list_of_audio_files = []
+        for _ in range(self._seconds_to_check_for_audio_files):
+            list_of_audio_files = glob.glob(os.path.join(directory, "*." + extension))
+            if len(list_of_audio_files) > 0 and \
+                    any(file_prefix in string for string in list_of_audio_files):
+                break
+            time.sleep(0.2)
+        for file in list_of_audio_files:
+            if file.find(file_prefix) >= 0:
+                return file
+        rospy.loginfo("No matching files found, returning None")
+        return None
+
+
+if __name__ == "__main__":
+    rospy.init_node("reading_evaluator")
+    DATABASE_NAME = "vision-project"
+    host = rospy.get_param("mongodb/host")
+    port = rospy.get_param("mongodb/port")
+    state_database = StateDb(
+        pymongo.MongoClient(host, port),
+        database_name=DATABASE_NAME,
+        collection_name="state_db"
+    )
+    init_db(state_database, INITIAL_STATE_DB)
+
+    source_directory = rospy.get_param("vision-project/data/data_capture")
+
+    reading_evaluator = ReadingEvaluator(
+        silence_threshold=-40,
```

Comment:
The units of these arguments aren't obvious to me. Maybe change the names or add comments in the class?

Reply:
I've decided to not include the speech recognition module because that requires setting up accounts with cloud services that have a free tier limit, so I removed this parameter.